stoobly_agent/cli.py

```diff
@@ -44,11 +47,39 @@ def run(**kwargs):
     settings.proxy_url = f"http://{kwargs['proxy_host']}:{kwargs['proxy_port']}"
     settings.agent_url = f"http://{kwargs['ui_host']}:{kwargs['ui_port']}"
 
+    if kwargs['api_url']:
+        settings.api_url = kwargs['api_url']
+
     if not kwargs['headless']:
         initialize_ui(kwargs)
 
     initialize_proxy(kwargs)
 
+@main.command()
+@click.option('--pretty-print', is_flag=True, default=False, help='Pretty print the json.')
+@click.option('--save-to-file', is_flag=True, default=False, help='To save to a file or not.')
+def dump_config(**kwargs):
+    settings = Settings.instance()
+    settings_dict = settings.__dict__
+    settings_dict['is_headless'] = settings.is_headless()
+    settings_dict['is_debug'] = settings.is_debug()
+    output = None
+
+    if kwargs['pretty_print']:
+        output = json.dumps(settings_dict, indent=4)
+    else:
+        output = json.dumps(settings_dict)
+
+    print(output)
+
+    if kwargs['save_to_file']:
```

Comment:
Maybe abstract this to Settings.instance().save_to_file()

Reply:
Done in https://github.com/Stoobly/stoobly-agent/pull/6/commits/bae8ae7ed9d4de7deab22ab65f3f07479433c22f